core/src/test/java/kafka/test/api/ShareConsumerTest.java

```diff
@@ -1922,6 +1915,37 @@ private void warmup() throws InterruptedException {
         }
     }
 
+    private void verifyShareGroupStateTopicRecordsProduced() {
+        try {
+            TestUtils.waitForCondition(() ->
```

Comment:
This condition will already be satisfied because it mirrors the `warmup()` method which runs before each test. I think you can remove this as redundant code.